js.go

```diff
@@ -1357,6 +1357,9 @@ func checkConfig(s, u *ConsumerConfig) error {
 	if u.Replicas > 0 && u.Replicas != s.Replicas {
 		return makeErr("replicas", u.Replicas, s.Replicas)
 	}
+	if u.MemoryStorage && !s.MemoryStorage {
```

Comment:
I think we should be verifying it both ways, i.e. `true -> false` and `false -> true`. Right now check will pass even if `u.MemoryStorage == false` and `s.MemoryStorage == true`.
```suggestion
	if u.MemoryStorage != !s.MemoryStorage {
```

Reply:
No we can't unfortunately, because we don't have a "3 states" value: true, false, not-specified. The point of this function is to check the user *intent* vs what is already configured in the server. 
We have no way to express that the user does not want MemoryStorage, so if you call js.Subscribe() without specifying anything and the JS consumer happens to be configured on the server with MemoryStorage, you don't want to call to fail. Otherwise we would need an option that says "NoMemoryStorage", etc..

So this function for boolean is not great, but the best we can do is check when the intent was to set a boolean to true, then only we can check that it matches the server setting.